### PR TITLE
refactor: input budget field to include min/max per day and week

### LINKED ISSUE #1595: Change format of sending Budget goals to scheduler
**Is your feature request related to a problem? Please describe.**
The way we define Budget has changed.  
A Budget now always includes an min-max per day and a min-max per week.

**Describe the solution you'd like**
Whenever there is a Budget to send, send this to the scheduler:
```
"budget": {
  "minPerDay": 2,
  "maxPerDay": 10,
  "minPerWeek": 40,
  "maxPerWeek": 40
},
```
...instead of sending this:  
```
"budgets": [{
    "budget_type": "Weekly",
    "min": 40
}]
```

**Describe alternatives you've considered**
None.

**Additional context**
This is not a bug. We just changed the approach.

### PR DESCRIPTION
**Pull request by vinaybadgujar102**
resolves #1595

### CODE REVIEW

**Review by tijlleenders — CHANGES_REQUESTED** [review]
I don't see any goals/budgets being sent to the scheduler...
image

**Comment by vinaybadgujar102** [comment]
as mentioned in #1594, not all entities are currently being sent to the scheduler. 
You can create new goals with duration and check the budget structure.

**Comment by tijlleenders** [comment]
@vinaybadgujar102 
I understand something without a duration isn't being sent. Budgets are an exception to this rule. Maybe that was an oversight.

The default goals are budgets - so I shouldn't have to create anything for them to be sent to the scheduler.

**Comment by tijlleenders** [comment]
Added #1674 to require budgets being sent.

**Comment by tijlleenders** [comment]
Looks good I think.
@vinaybadgujar102 We're now just sending everything, right?

@Tushar-4781 Agree on code?

**Comment by vinaybadgujar102** [comment]
Yeah, we are sending everything to the scheduler

File: src/Interfaces/IScheduler.ts [hunk 1]
```diff
@@ -39,7 +39,12 @@ export interface ISchedulerInputGoal {
     not_on?: blockedSlotOfTask[];
   };
   repeat?: string;
-  budgets?: { budget_type: "Daily" | "Weekly"; min?: number; max?: number }[];
+  budget?: {
+    minPerDay: number | undefined;
+    maxPerDay: number | undefined;
```
Comment (Tushar-4781) [inline src/Interfaces/IScheduler.ts:44]: do like this 
minPerDay?: number

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]

**Review by Tushar-4781 — APPROVED** [review]

**Comment by Tushar-4781** [comment]
@vinaybadgujar102 please resolve the conflicts so that we can merge :)

**Review by tijlleenders — APPROVED** [review]

**Pull request merged**